### PR TITLE
Tagged Metrics

### LINKED ISSUE #80: Prebid Server account showing Requests Per Second and Bids Per Second in minus
In Prebid Server account (https://prebid.adnxs.com/account/) Requests Per Second and Bids Per Seconds showing in minus values.

rpm_in_minus_censored

### PR DESCRIPTION
**Pull request by dbemiller**
Two big motivations here:

- Make #80 fixable (adding `hostname` tag so we can distinguish between deployed instances)
- Use Tags in the schema, [as Influx suggests](https://docs.influxdata.com/influxdb/v1.1/concepts/schema_and_data_layout/#don-t-encode-data-in-measurement-names)

Since the latter required quite a bit of code (details below), I split it into a new module to avoid flooding `pbs_light.go`. I structured it with the following ideas in mind, since they seemed valuable and weren't hard to support.

- Use DB-agnostic APIs
- Make the code unit-testable

---------------------------

It looks like there's a lot of tension between golang, metrics, and interfaces. Influx supports (and prefers) Tags for measurement metadata, but go-metrics [is dragging their feet on support](https://github.com/rcrowley/go-metrics/issues/135#issuecomment-278751054). With good reason, since their core APIs can't handle tags seamlessly.

I also looked into [go-kit](https://github.com/go-kit/kit/tree/master/metrics), but was disappointed to see that their APIs only support floats, and their Histograms [stop at p99](https://github.com/go-kit/kit/blob/master/metrics/influx/influx.go#L152). That seems less precise than we'd want for a major real-time system like this.

So... this implements a bastardized version which isn't clean enough to contribute to the go-metrics project, but does get the job done for our uses. It basically takes [this guy's strategy](https://github.com/rcrowley/go-metrics/issues/135#issuecomment-302259572), except updated to use v2 of the [Influx client](https://github.com/influxdata/influxdb/tree/master/client/v2).

After merging this PR, the next steps in the plan are:

1. Translate & validate any queries we need to use the Tagged Measurements. Since I added a `hostname` tag, we should be able to use those to fix #80.
2. Update all our queries on dashboards and the website to use the new metrics.
3. Merge #84 to delete the old metrics.

### CODE REVIEW

**Comment by mkendall07** [comment]
@asweeney86 want to review this?

File: metrics/pbsmetrics.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// RequestSource is the list of sources where requests might come from.
+// This obviously isn't comprehensive... just defined on an as-needed basis.
+type RequestSource int
+
+const (
+	// Safari has restrictive policies on 3rd party cookies. This helps measure how much of an effect that has.
+	SAFARI RequestSource = iota
+	APP
+	OTHER
+	UNKNOWN
```
Comment (mkendall07) [inline metrics/pbsmetrics.go]: Let's add `desktop` and make it the default. Also Safari doesn't seem to fit here if the idea is platform.

File: metrics/influx/influx.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+// These are the names of Measurements which we export to Influx.
+const (
+	AUCTION_REQUEST_COUNT    = "prebidserver.auction_request_count"
+	AUCTION_RESPONSE_COUNT   = "prebidserver.auction_response_count"
+	AUCTION_REQUEST_DURATION = "prebidserver.auction_request_duration_nanos"
+
+	BIDDER_REQUEST_COUNT    = "prebidserver.bidder_request_count"
+	BIDDER_RESPONSE_COUNT   = "prebidserver.bidder_response_count"
+	BIDDER_REQUEST_DURATION = "prebidserver.bidder_request_duration_nanos"
+	BID_PRICES              = "prebidserver.bid_response_cpm_cents"
```
Comment (mkendall07) [inline metrics/influx/influx.go]: Not seeing any data recorded against `prebidserver.bid_response_cpm_cents` or `prebidserver.bid_count` measurements.

Comment (dbemiller) [inline metrics/influx/influx.go]: Good catch... should be fixed now.

File: metrics/pbsmetrics.go (hunk 1 above)
Comment (dbemiller) [inline metrics/pbsmetrics.go]: I think you're right... I mucked up that data model.

`Browser` and `Source` are now separate tags. `Source` can be `desktop` or `app`. `Browser` can be `safari` or `unknown`.

**Pull request merged**